nrf_security/cmake/kconfig_mbedtls_configure.cmake

```diff
@@ -82,9 +82,10 @@ macro(mbedtls_config_define_depends mbedtls_config)
   set(${mbedtls_config} TRUE)
   foreach(arg ${ARGN})
     if (NOT (CONFIG_${arg} OR
-             CONFIG_CC310_${arg} OR
+             CONFIG_NRF_CRYPTO_BACKEND_COMBINATION_0 AND
+             (CONFIG_CC310_${arg} OR
              CONFIG_OBERON_${arg} OR
-             CONFIG_VANILLA_${arg}))
+             CONFIG_VANILLA_${arg})))
```

Comment:
If any of: `CONFIG_CC310_${arg}`, `CONFIG_OBERON_${arg}`, or `CONFIG_VANILLA_${arg}` can be set when `CONFIG_NRF_CRYPTO_BACKEND_COMBINATION_0` is false, then we have an error somewhere in our Kconfig, and that error should be fixed.

Reply:
No. That was the case from before, but we added the concept of "shared" code to ensure all features are available even though a backend that doesn't give access to the feature. An example of this is SHA-512 which is only available from the vanilla backend.

Instead of creating a duplicate cmake variable to address the need to enable compilation of the sha512.c file, we went for an approach that uses just one single config-variable. 

This is the main reason for this issue and th, which the additional call to test CONFIG_NRF_CRYPTO_BACKEND_COMBINATION_0 fixes

Reply:
We found a better scheme than the proposed method which didn't misrepresent the kconfig variables